archive.c

```diff
@@ -524,7 +524,7 @@ archive_find_file (ArchiveIndex *archive_index, const char *path,
     }
     current = current->next;
     if (*position + file->fsstat.file_size >= (u64) stat_buf.st_size) {
-      *position = 0x50;
+      *position = 0x50 + (*position + file->fsstat.file_size) % stat_buf.st_size;
```

Comment:
Ha! That's the bug that caused the extracted files to seem corrupted sometimes! :+1: